python/cudf/dataframe/dataframe.py

```diff
@@ -281,7 +281,7 @@ def assign(self, **kwargs):
              2    2    5
 
         """
-        new = self.copy()
+        new = self.copy(deep=True)
```

Comment:
Does pandas assign by copy as opposed to assign by reference by default?

Reply:
Hmm, this is a good catch. No, Pandas demonstrates in `test_pandas_dataframe_copy()` that `pdf.copy()` is shallow: changing `copy_pdf` always changes `pdf` also.

This is a problem. I guess I've gone back and forth on this now twice because originally I set `df.copy()` to make a shallow copy. This broke the main unit tests in `dataframe.py`. I dreamed of submitting the PR, not rewriting six core methods (4 maybe?) in `dataframe.py` and happily changed default to Deep...

Reply:
Unfortunately I will have to fix the core functionality if we're to make `.copy()` default to shallow. I can share the failing tests with you, unless you want to pull it and build it yourself. I'm mostly done for the night so I'll do that in the morning.